content/en/blog/_posts/2023-09-15-bootstrap-an-air-gapped-cluster-with-kubeadm/index.md

```diff
@@ -0,0 +1,681 @@
+
+DOCKER_USER=user
+DOCKER_PASS=pass
+echo ${DOCKER_PASS} | docker login --username=${DOCKER_USER} --password-stdin localhost:5000
+```
+
+Create a cluster configuration file and initialize the cluster:
+
+```bash
+echo "---
+
+apiVersion: kubeadm.k8s.io/v1beta3
+kind: ClusterConfiguration
+clusterName: kubernetes
+kubernetesVersion: v1.27.3
+networking:
+    dnsDomain: cluster.local
+    podSubnet: 10.244.0.0/16 # --pod-network-cidr
+    serviceSubnet: 10.96.0.0/12
+---
+apiVersion: kubeadm.k8s.io/v1beta3
+kind: InitConfiguration
+localAPIEndpoint:
+    advertiseAddress: 10.10.10.10 # Update to the IP address of the air gap VM
+    bindPort: 6443
+nodeRegistration:
+    criSocket: unix:///run/containerd/containerd.sock # or rely on autodetection
+    name: airgap # this must match the hostname of the air gap VM
+# Since this is a single node cluster, this taint has to be commented out,
+# otherwise the coredns pods will not come up.
+# taints:
+# - effect: NoSchedule
+# key: node-role.kubernetes.io/master" > kubeadm_cluster.yaml
+
+kubeadm init --config kubeadm_config.yaml
+```
+
+Set `$KUBECONFIG` and use `kubectl` to wait until the API server is healthy:
+
+```bash
+export KUBECONFIG=/etc/kubernetes/admin.conf
+
+until kubectl get nodes; do
+    echo -e "\nWaiting for API server to respond..." 1>&2
+    sleep 5
+
+done
+```
+
+### Set up networking
+
+Update Flannel image locations in the Flannel manifest, and apply it:
+
+```bash
+sed -i 's/image: docker\.io/image: localhost:5000/g' kube-flannel.yaml
+kubectl apply -f kube-flannel.yaml
+```
+
+Run `kubectl get pods -A --watch` until all pods are up and running.
+
+## Run an example Pod
+
+With a cluster operational, the next step is a workload. For this simple demonstration, the [Podinfo](https://github.com/stefanprodan/podinfo) application will be deployed.
+
+### Install Helm
+
+This first part of the procedure must be executed from the host/laptop. If not already present, install Helm following [Installing Helm](https://helm.sh/docs/intro/install/).
+
+Next, download the helm binary for Linux:
+
+```bash
+UARCH=$(uname -m)
+# Reset the architecture variables if needed
+if [["$UARCH" == "arm64" || "$UARCH" == "aarch64"]]; then
+
+    ARCH="aarch64"
+    K8s_ARCH="arm64"
+
+else
+
+    ARCH="x86_64"
+    K8s_ARCH="amd64"
+
+fi
+
+curl -LO https://get.helm.sh/helm-v3.12.2-linux-${K8s_ARCH}.tar.gz
+```
+
+Add the Podinfo helm repository, download the Podinfo helm chart, download the Podinfo container image, and then finally save it to the local disk:
+
+```bash
+helm repo add https://stefanprodan.github.io/podinfo
+helm fetch podinfo/podinfo --version 6.4.0
+docker pull ghcr.io/stefanprodan/podinfo:6.4.0
+```
+
+**Save the image to a tar file on the local disk**
+```bash
+docker save -o podinfo_podinfo-6.4.0.tar ghcr.io/stefanprodan/podinfo
+```
+**Reuse the ~/tmp directory created on the air gapped VM to transport these artifacts across the air gap:**
```

Comment:
```suggestion

```suggestion

### Transfer the image across the air gap

Reuse the `~/tmp` directory created on the air gapped VM to transport these artifacts across the air gap:

```